test/sharness/t0087-repo-robust-gc.sh

```diff
@@ -37,6 +39,23 @@ test_gc_robust_part1() {
 		ipfs repo gc
 	'
 
+	test_expect_success "corrupt the root node of 1MB file" '
+		ls -l "$HASH1FILE"
+		test -e "$HASH1FILE" &&
+		dd if=/dev/zero of="$HASH1FILE" count=1 bs=100 conv=notrunc
+	'
+
+	test_expect_success "'ipfs repo gc' should abort without removing anything" '
+		test_must_fail ipfs repo gc 2>&1 | tee gc_err &&
+		grep -q "could not retrieve links for $HASH1" gc_err &&
+		grep -q "aborted" gc_err
+	'
+
+	test_expect_success "leaf nodes where not removed after gc" '
+		ipfs cat $LEAF3 > /dev/null &&
+		ipfs cat $LEAF4 > /dev/null
```

Comment:
Is this test run both with and without a daemon? If yes, we should add a `--timeout` flag here, if no, we should make sure to run this test with and without a daemon ;)